adapters/ddf-adapter/src/test/java/com/connexta/replication/adapters/ddf/DdfNodeAdapterTest.java

```diff
@@ -0,0 +1,297 @@
+import org.codice.ddf.cxf.client.ClientFactoryFactory;
+import org.codice.ddf.cxf.client.SecureCxfClientFactory;
+import org.codice.ditto.replication.api.AdapterException;
+import org.codice.ditto.replication.api.data.Metadata;
+import org.codice.ditto.replication.api.data.QueryRequest;
+import org.codice.ditto.replication.api.data.QueryResponse;
+import org.codice.ditto.replication.api.data.Resource;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DdfNodeAdapterTest {
+  @Mock DdfRestClientFactory ddfRestClientFactory;
+  @Mock ClientFactoryFactory clientFactory;
+
+  @Mock DdfRestClient restClient;
+
+  @Mock SecureCxfClientFactory secureFactory;
+
+  @Mock Csw csw;
+
+  private DdfNodeAdapter adapter;
+
+  @Before
+  public void setUp() throws Exception {
+    when(ddfRestClientFactory.create(any(String.class))).thenReturn(restClient);
+    when(clientFactory.getSecureCxfClientFactory(
+            any(String.class),
+            any(Class.class),
+            any(List.class),
+            any(Interceptor.class),
+            any(Boolean.class),
+            any(Boolean.class),
+            any(Integer.class),
+            any(Integer.class)))
+        .thenReturn(secureFactory);
+    when(secureFactory.getClient()).thenReturn(csw);
+    adapter =
+        new DdfNodeAdapter(
+            ddfRestClientFactory, clientFactory, new URL("https://localhost:8994/searvice"));
+  }
+
+  @Test
+  public void isAvailable() throws Exception {
+    CapabilitiesType response = mock(CapabilitiesType.class);
+    when(csw.getCapabilities(any(GetCapabilitiesType.class))).thenReturn(response);
+    when(response.getVersion()).thenReturn("2.0.2");
+    assertThat(adapter.isAvailable(), is(true));
+  }
+
+  @Test
+  public void isAvailableError() throws Exception {
+    when(csw.getCapabilities(any(GetCapabilitiesType.class))).thenThrow(new Exception("error"));
+    assertThat(adapter.isAvailable(), is(false));
+  }
+
+  @Test
+  public void getSystemName() throws Exception {
+    CswRecordCollection collection = new CswRecordCollection();
+    Metadata metadata = getMetadata();
+    collection.setCswRecords(Collections.singletonList(metadata));
+    when(csw.getRecords(any(GetRecordsType.class))).thenReturn(collection);
+    assertThat(adapter.getSystemName(), is("mytitle"));
+  }
+
+  @Test
+  public void getSystemNameCached() throws Exception {
+    CswRecordCollection collection = new CswRecordCollection();
+    Metadata metadata = getMetadata();
+    collection.setCswRecords(Collections.singletonList(metadata));
+    when(csw.getRecords(any(GetRecordsType.class))).thenReturn(collection);
+    assertThat(adapter.getSystemName(), is("mytitle"));
+    assertThat(adapter.getSystemName(), is("mytitle"));
+    verify(csw, times(1)).getRecords(any(GetRecordsType.class));
```

Comment:
❓ Why are we calling the CSW client when the system name is cached? This seems incorrect. Should this instead be asserting 0 interactions?

Reply:
The first call does hit the csw client but the second one does not so with two requests we should only hit csw once.